File: chat_kit_sample/src/main/java/com/zoho/deskportalsdk/demo/chat_kit_sample/MainActivity.java
package com.zoho.deskportalsdk.demo.chat_kit_sample;

import android.os.Bundle;
import android.view.View;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.zoho.desk.asap.chatkit.ZohoDeskPortalChatKit;

import java.util.ArrayList;
import java.util.HashMap;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void showGC(View view) {
        ZohoDeskPortalChatKit.showGC(this);
    }
    public void showAnswerBot(View view) {
        ZohoDeskPortalChatKit.showAnswerBot(this);
    }
    public void showBM(View view) {
        ZohoDeskPortalChatKit.showBM(this);
    }

    public void setGCSessionVariable(View view) {
        ArrayList<HashMap<String, Object>> sessionVariables = new ArrayList<>();

        HashMap<String, Object> firstVariable = new HashMap<>();
        firstVariable.put("name", "YOUR VARIABLE NAME");
        firstVariable.put("value", "YOUR VARIABLE VALUE");

        HashMap<String, Object> secondVariable = new HashMap<>();
        secondVariable.put("name", "YOUR SECOND VARIABLE NAME");
        secondVariable.put("value", "YOUR SECOND VARIABLE VALUE");

        sessionVariables.add(firstVariable);
        sessionVariables.add(secondVariable);

        ZohoDeskPortalChatKit.setGCSessionVariable(this, sessionVariables);
    }

    public void updateGCSessionVariable(View view) {
        ArrayList<HashMap<String, Object>> sessionVariables = new ArrayList<>();

        HashMap<String, Object> firstVariable = new HashMap<>();
        firstVariable.put("name", "YOUR VARIABLE NAME");
        firstVariable.put("value", "YOUR VARIABLE VALUE");

        HashMap<String, Object> secondVariable = new HashMap<>();
        secondVariable.put("name", "YOUR SECOND VARIABLE NAME");
        secondVariable.put("value", "YOUR SECOND VARIABLE VALUE");

        sessionVariables.add(firstVariable);
        sessionVariables.add(secondVariable);

        ZohoDeskPortalChatKit.updateGCSessionVariable(this, sessionVariables);
    }

    public void setBMSessionVariable(View view) {
        ArrayList<HashMap<String, Object>> sessionVariables = new ArrayList<>();

        HashMap<String, Object> firstVariable = new HashMap<>();
        firstVariable.put("name", "YOUR VARIABLE NAME");
        firstVariable.put("value", "YOUR VARIABLE VALUE");

        HashMap<String, Object> secondVariable = new HashMap<>();
        secondVariable.put("name", "YOUR SECOND VARIABLE NAME");
        secondVariable.put("value", "YOUR SECOND VARIABLE VALUE");

        sessionVariables.add(firstVariable);
        sessionVariables.add(secondVariable);

        ZohoDeskPortalChatKit.setBMSessionVariable(this, sessionVariables);
    }

    public void updateBMSessionVariable(View view) {
        ArrayList<HashMap<String, Object>> sessionVariables = new ArrayList<>();

        HashMap<String, Object> firstVariable = new HashMap<>();
        firstVariable.put("name", "YOUR VARIABLE NAME");
        firstVariable.put("value", "YOUR VARIABLE VALUE");

        HashMap<String, Object> secondVariable = new HashMap<>();
        secondVariable.put("name", "YOUR SECOND VARIABLE NAME");
        secondVariable.put("value", "YOUR SECOND VARIABLE VALUE");

        sessionVariables.add(firstVariable);
        sessionVariables.add(secondVariable);

        ZohoDeskPortalChatKit.updateBMSessionVariable(this, sessionVariables);
    }

}

File: chat_kit_sample/src/main/java/com/zoho/deskportalsdk/demo/chat_kit_sample/MainActivity.kt
package com.zoho.deskportalsdk.demo.chat_kit_sample

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.zoho.desk.asap.livechat.ZDPortalLiveChat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun showGC(view: View) {
        ZDPortalLiveChat.showGC(this)
    }
    fun showAnswerBot(view: View) {
        ZDPortalLiveChat.showAnswerBot(this)
    }
    fun showBM(view: View) {
        ZDPortalLiveChat.showBusinessMessenger(this)
    }

    fun setGCSessionVariable(view: View) {
        val sessionVariables = arrayListOf(
            hashMapOf<String, Any>("name" to "YOUR VARIABLE NAME", "value" to "YOUR VARIABLE VALUE"),
            hashMapOf<String, Any>("name" to "YOUR SECOND VARIABLE NAME", "value" to "YOUR VARIABLE VALUE")
        )
        ZDPortalLiveChat.setGCSessionVariable(this, sessionVariables)
    }

    fun updateGCSessionVariable(view: View) {
        val sessionVariables = arrayListOf(
            hashMapOf<String, Any>("name" to "YOUR VARIABLE NAME", "value" to "YOUR VARIABLE VALUE"),
            hashMapOf<String, Any>("name" to "YOUR SECOND VARIABLE NAME", "value" to "YOUR VARIABLE VALUE")
        )
        ZDPortalLiveChat.updateGCSessionVariable(this, sessionVariables)
    }

    fun setBMSessionVariable(view: View) {
        val sessionVariables = arrayListOf(
            hashMapOf<String, Any>("name" to "YOUR VARIABLE NAME", "value" to "YOUR VARIABLE VALUE"),
            hashMapOf<String, Any>("name" to "YOUR SECOND VARIABLE NAME", "value" to "YOUR VARIABLE VALUE")
        )
        ZDPortalLiveChat.setBMSessionVariable(this, sessionVariables)
    }

    fun updateBMSessionVariable(view: View) {
        val sessionVariables = arrayListOf(
            hashMapOf<String, Any>("name" to "YOUR VARIABLE NAME", "value" to "YOUR VARIABLE VALUE"),
            hashMapOf<String, Any>("name" to "YOUR SECOND VARIABLE NAME", "value" to "YOUR VARIABLE VALUE")
        )
        ZDPortalLiveChat.updateBMSessionVariable(this, sessionVariables)
    }

}
